lib/fluent/plugin/out_s3.rb

```diff
@@ -630,6 +631,36 @@ def compress(chunk, tmp)
       end
     end
 
+    class ZstdCompressor < Compressor
+      require 'zstd-ruby'
+      
+      DEFAULT_LEVEL = 3
+    
+      def initialize(options = {})
+        @level = (options.is_a?(Hash) && options[:level]) || DEFAULT_LEVEL
+      end
+    
+      def ext
+        'zst'.freeze
+      end
+    
+      def content_type
+        'application/x-zst'.freeze
+      end
+    
+      def compress(chunk, tmp)
+        begin
+          data = chunk.read.gsub(/\r\n/, "\n").force_encoding('UTF-8')
```

Comment:
> `data = chunk.read.gsub(/\r\n/, "\n").force_encoding('UTF-8')`

Could you please tell me why this is necessary?

Reply:
https://github.com/fluent/fluent-plugin-s3/pull/439#discussion_r1827143270

> Handled line endings and encoding

Could you please add this test case?

Reply:
@daipom 
I've found that those lines `(.gsub(/\r\n/, "\n").force_encoding('UTF-8')) `were mistakenly added during my initial troubleshooting of the compression issues. 

The real issue was not with line endings or encoding, but with the compression implementation itself. After fixing the compression logic to use `chunk.write_to` method properly, the compression and tests are now working correctly without any need for explicit line ending or encoding manipulation.

I'll remove those unnecessary lines since they don't serve any purpose in the proper implementation.